### DISCUSSION TITLE
List users and enable pending via command line (bearer token)?

### DISCUSSION THREAD

**Discussion by stevenbaert** [open-webui/open-webui#6254] [Q&A]
Hi,

Was looking for a way to list open-webui users and enable them via command line. That way I could be informed by mail a new user logged in and approve or refuse access (delete) in automated way.

This would be an approach but did not find any bearer-token or way to set one in container(?)
---

headers = {
    'Authorization': 'Bearer YOUR_TOKEN',
    'Content-Type': 'application/json'
}

url = "https://mydomain/api/users"

response = requests.get(url, headers=headers)

if response.status_code == 200:
    users = response.json()
    for user in users:
        print(user)
else:
    print("Error:", response.json())

Thanks!
S.

**Comment by oboudry** [comment]
Here is how you enable API Keys:

- Set the ENABLE_API_KEYS variable to true
- Make sure user/group has permissions to use API (even if admin)
- Go to Settings / Account to create your API Key.

And here is how you get all users:
```
curl -sSL -H "Authorization: Bearer sk-YOURKEYHERE" -H "Accept: application/json" https://open-webui.k8s-lab.group/api/v1/users/all
```

I tried to enable API keys from the UI but it did not work on a setup running multiple nodes. Environment variable sorted it.
